test/csv/parse/test_convert.rb

```diff
@@ -107,4 +107,45 @@ def test_empty_value
     assert_equal([nil, "empty", "a"],
                  CSV.parse_line(',"",a', empty_value: "empty"))
   end
+
+  sub_test_case("add quoted info") do
+    def setup
+      @preserving_converter = lambda do |field, info|
+        f = field.encode(CSV::ConverterEncoding)
+        return field if info.quoted?
+        return DateTime.parse(f) if f.match?(DateTimeMatcher) rescue
+        return Integer(f) rescue
+        return Float(f) rescue
+        field
```

Comment:
Can we simplify this?

```suggestion
        f = field.encode(CSV::ConverterEncoding)
        return f if info.quoted?
        Integer(f, 10)
```